CHAID/CHAID.py

```diff
@@ -431,14 +481,31 @@ def generate_best_split(self, ind, dep):
                 index[index.arr == choice[1]] = choice[0]
                 unique.remove(choice[1])
 
-                for val, count in frequencies[choice[1]].items():
-                    frequencies[choice[0]][val] += count
-                del frequencies[choice[1]]
+                for val, count in freq[choice[1]].items():
+                    freq[choice[0]][val] += count
+                del freq[choice[1]]
+
+                if wt is not None:
```

Comment:
HA! Making everyone the same weight isn't the same as not using weights. Retarded, I know.